src/cegpy/trees/staged.py

```diff
@@ -446,25 +448,37 @@ def _mark_nodes_with_stage_number(self, merged_situations) -> list:
 
         return list_of_merged_situations
 
-    def _generate_colours_for_situations(self, merged_situations):
+    def _generate_colours_for_situations(self, merged_situations, colour_list):
         """Colours each stage of the tree with an individual colour"""
         number_of_stages = len(merged_situations)
-        stage_colours = Util.generate_colours(number_of_stages)
+        if colour_list is None:
+            stage_colours = Util.generate_colours(number_of_stages)
+        else: 
+            stage_colours = colour_list
+            if len(colour_list) < number_of_stages:
+                logger.warning(
+                    "The number of colours is less than the number" +
+                    "of stages. Colours will be recycled."
+                )
         self._stage_colours = stage_colours
-
+        iter_colour = cycle(stage_colours)
         for node in self.nodes:
             try:
                 stage = self.nodes[node]['stage']
-                self.nodes[node]['colour'] = stage_colours[stage]
+                self.nodes[node]['colour'] = next(iter_colour)
```

Comment:
I like this, very pythonic

Reply:
Infact, I will modify Util.generate_colours() to be a generator function once this has been merged.